### ISSUE TITLE
sort Alphabetical <a href="my_dynamic_link">values</a>

### ISSUE THREAD

**Issue by Treenity** [wenzhixin/bootstrap-table#461]
hi, 

i'm sorry but i really don't know javascript, so i'm going to explain :
i need to sort a column containing names with link codes.

here is what i have

Name / profil_id

my table need to show only names(sortables) but the formatter will return : 
<a href="http://www......index.php?link_id=(the profil_id)">name</a> <----- link pointing to his profile

i don't know how to do that, maybe microdata in <td>, or regex from javascript formatting code, or maybe is there already a function to do that?

thanks

**Comment by Treenity** [comment]
ok i've found a alternativ, i give : 
name[profil_id]

then with data-formatter and a regex i return the formated url

**Comment by wenzhixin** [comment]
i think you can use the sorter column option to do that.

**Comment by melissaltaylor** [comment]
Is there a working example of this? I am trying to sort based on an attribute on the tds (alphabetically) but it keeps reading all the html tags in the cell.

**Comment by wenzhixin** [comment]
http://issues.wenzhixin.net.cn/bootstrap-table/#options/custom-sort.html @melissaltaylor

**Comment by melissaltaylor** [comment]
Thanks for responding so quickly, but the example doesn't show an attribute on the td's specifically, just sorting option on the th's. This is more what I am looking for - to be able to sort by the 'data-title' values on the tds below:
`<table data-toggle=“table”>
  <thead>
    <tr>
      <th>Book Preview</th>
      <th data-sortable="true" data-sort-name="data-title">Description</th>
      <th data-sortable="true">Year Published</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src=“./images/alice.png“></td>
      <td data-title=“Alice in Wonderland”>
        <a class=“download” href=“alice.pdf”>
          <h1>Alice in Wonderland</h1>
          <h2>Lewis Carroll</h2>
        </a>
      </td>
      <td>1865</td>
    </tr>
    <tr>
      <td><img src=“./images/winnie.png“></td>
      <td data-title=“Winnie the Pooh”>
        <a class=“download” href=“pooh.pdf”>
          <h1>Winnie the Pooh</h1>
          <h2>A. A. Milne</h2> 
        </a>
      </td>
      <td>1926</td>
    </tr>
    <tr>
      <td><img src=“./images/catinhat.png”></td>
      <td data-title=“The Cat in the Hat”>
        <a class=“download” href=“catinhat.pdf”>
          <h1>The Cat in the Hat</h1>
          <h2>Dr. Seuss</h2>
        </a>
      </td>
      <td>1957</td>
    </tr>
  </tbody>
</table>`

**Comment by wenzhixin** [comment]
pls provide a jsfiddle, maybe I can help you.

**Comment by melissaltaylor** [comment]
Here is the fiddle (part 1):

http://jsfiddle.net/e1ypfk3b/1/

It looks like the sorting for column Title is happening on the <a href='x'>
information rather than on the text in the cell. So I tried adding an
attribute to the td to pull out the information I want the rows to sort by.

I tried adding an additional data-title="Alice in Wonderland" (etc.) within
the td, but the sorting is still not working.

http://jsfiddle.net/871c89sh/6/
